ui/dashboards/src/components/TimeRangeControls/TimeRangeControls.tsx

```diff
@@ -39,13 +34,8 @@ export const TIME_OPTIONS: TimeOption[] = [
 
 export function TimeRangeControls() {
   const { timeRange, setTimeRange } = useTimeRange();
-  const dashboardDefaultTimeRange = useDefaultTimeRange();
-  const { queryString } = useQueryString();
-
-  const defaultTimeRange = getDefaultTimeRange(dashboardDefaultTimeRange, queryString);
-
-  // selected form value can be relative or absolute, timeRange from plugin-system is only absolute
-  const [selectedTimeRange, setSelectedTimeRange] = useState<TimeRangeValue>(defaultTimeRange);
+  const { selectedTimeRange } = useSelectedTimeRange();
```

Comment:
I think I mentioned this in the previous PR, but as I'm reading this, I'm a little confused as to what the difference is between `selectedTimeRange` and the `timeRange` directly above this line? `E_TOO_MANY_TIME_RANGES` 😂 

Do we actually need both? Can one just be derived from the other?

Reply:
So I have been gradually simplifying it, in my first PR TimeRangeControls was even harder to follow 😣. I just updated this PR and was able to remove selectedTimeRange! 

This involved refactoring the plugin runtime so the isRelativeTimeRange conversion happens in useTimeRange. Then I created an "internal" version of this API in the dashboards package (see useDashboardTimeRange) that supports relative and absolute time ranges (since the plugin-system version is for [resolved absolute values](https://github.com/perses/perses/pull/530#discussion_r972423337)).